### ISSUE TITLE
Add comprehensive documentation to lib.rs

### ISSUE THREAD

**Issue by joshrotenberg** [joshrotenberg/docker-wrapper#55]
## Summary
Add comprehensive crate-level documentation to lib.rs including getting started guide, usage examples, and feature documentation.

## Motivation
The lib.rs file currently has minimal documentation. Users need comprehensive crate-level documentation to understand:
- How to get started with the crate
- Common usage patterns
- Available features and how to enable them
- Architecture overview
- Best practices

### 1. Crate Overview
- Purpose and goals of docker-wrapper
- Key features and benefits
- Comparison with alternatives

### 2. Getting Started
```rust
//! ## Getting Started
//! 
//! Add docker-wrapper to your Cargo.toml:
//! ```toml
//! [dependencies]
//! docker-wrapper = "0.2"
//! ```
//! 
//! For Docker Compose support:
//! ```toml
//! [dependencies]
//! docker-wrapper = { version = "0.2", features = ["compose"] }
//! ```
```

### 3. Basic Usage Examples
```rust
//! ## Quick Start
//! 
//! Run a container:
//! ```rust
//! use docker_wrapper::{DockerCommand, RunCommand};
//! 
//! # async fn example() -> Result<(), Box<dyn std::error::Error>> {
//! let output = RunCommand::new("nginx:latest")
//!     .name("my-web-server")
//!     .port(8080, 80)
//!     .detach()
//!     .execute()
//!     .await?;
//! # Ok(())
//! # }
//! ```
```

### 4. Architecture Overview
- Command trait design
- Builder pattern implementation
- Error handling strategy
- Async/await usage

### 5. Feature Documentation
```rust
//! ## Features
//! 
//! - **Default**: Core Docker commands (run, ps, images, etc.)
//! - **compose**: Docker Compose support for multi-container applications
//! 
//! ### Using Docker Compose
//! ```rust
//! #[cfg(feature = "compose")]
//! use docker_wrapper::compose::{ComposeUpCommand, ComposeDownCommand};
//! ```
```

### 6. Common Patterns
- Container lifecycle management
- Image operations
- Monitoring and logging
- Error handling
- Resource cleanup

### 7. Command Coverage
Table or list showing all 35 implemented Docker commands with brief descriptions

### 8. Migration Guide
- Migrating from docker CLI scripts
- Migrating from other Docker crates

### 9. Safety and Best Practices
- Resource cleanup patterns
- Error handling
- Security considerations

## Implementation Tasks
- [ ] Write crate overview section
- [ ] Add getting started guide
- [ ] Include multiple usage examples
- [ ] Document all features
- [ ] Add architecture documentation
- [ ] Create command reference table
- [ ] Add migration guides
- [ ] Include best practices section
- [ ] Add links to examples directory
- [ ] Cross-reference with README

## Success Criteria
- [ ] New users can get started quickly
- [ ] All features are documented
- [ ] Common patterns are demonstrated
- [ ] Architecture is clearly explained
- [ ] Best practices are documented
- [ ] Examples compile and run
